tools/barista/src/transform.ts

```diff
@@ -89,30 +96,49 @@ export const extractH1ToTitleTransformer: BaPageTransformer = async source => {
 export const headingIdTransformer: BaPageTransformer = async source => {
   const transformed = { ...source };
   if (source.content && source.content.length) {
-    const content = loadWithCheerio(source.content, { xmlMode: true });
-    const headlines = content('h1, h2, h3, h4, h5, h6');
-    if (headlines.length) {
-      headlines.each((_, headline) => {
-        const text = content(headline).text();
-        const headlineId = text
-          .toLowerCase()
-          .replace(/&amp;/g, '')
-          .replace(/&/g, '')
-          .replace(/\?/g, '')
-          .replace(/\//g, '')
-          .replace(/’/g, '')
-          .replace(/:/g, '')
-          .replace(/\./g, '')
-          .replace(/,/g, '')
-          .replace(/\(/g, '')
-          .replace(/\)/g, '')
-          .replace(/”/g, '')
-          .replace(/[^\w&;:/?\(\)\.<>,’”]+/g, '-')
-          .replace(/^(\d+)/g, 'h$1');
-        content(headline).attr('id', headlineId);
-      });
-      transformed.content = content.html() || '';
-    }
+    transformed.content = runWithCheerio(source.content, $ => {
+      const headlines = $('h1, h2, h3, h4, h5, h6');
+      if (headlines.length) {
+        headlines.each((_, headline) => {
+          const text = $(headline).text();
+          const headlineId = text
+            .toLowerCase()
+            .replace(/&amp;/g, '')
```

Comment:
I guess this replace chain could also be written in a combined regex, but this is way easier to read 👍

Reply:
I think you wrote this part of the code, copied it from the smith ;-)